magritte/mesher.py

```diff
@@ -827,3 +835,158 @@ def get_recursive_remesh(positions: np.array, data: np.array, depth: int, max_de
+    origin : 1 by 3 numpy vector of float
+        Contains x,y,z coordinates of center point of the sphere (by default [0.0,0.0,0.0]^T)
+
+    Returns
+    -------
+    remeshed_positions : numpy array of float
+        The positions of the points in the remeshed point cloud. The boundary points lie in front.
+    number of boundary points : unsigned int
+        The number of boundary points in the remeshed point cloud.
+    '''
+
+    radii2 = lambda r : np.sum(np.power(r-origin[np.newaxis, :], 2),axis=1)#function for computing the square of the radius
+    #clear points within inner boundary
+    positions_reduced, nb_boundary = point_cloud_clear_inner_boundary_generic(remeshed_positions, nb_boundary, radii2, radius**2)
+    #use healpy with 12*5**2 directions to define inner sphere
+    N_inner_bdy = 12*healpy_order**2
+    #healpix always requires 12*x**2 as number of points on the sphere
+    direction  = healpy.pixelfunc.pix2vec(healpy.npix2nside(N_inner_bdy), range(N_inner_bdy))
+    direction  = np.array(direction).transpose()
+    #then multiply with the desired radius of the inner boundary
+    inner_bdy = origin + radius * direction
+    positions_reduced = np.concatenate((inner_bdy,positions_reduced))
+    nb_boundary = nb_boundary + N_inner_bdy
+
+    return positions_reduced, nb_boundary
+
+
+def point_cloud_add_spherical_outer_boundary(remeshed_positions, nb_boundary, radius, healpy_order = 10, origin = np.array([0.0,0.0,0.0]).T):
+    '''
+    Function for specifying a spherical outer boundary in a remeshed point cloud.
+    First clears the region outside the sphere and then constructs a spherical boundary using healpy.
+
+    Parameters
+    ----------
+    remeshed_positions : numpy array of float
+        Positions of the points in the point cloud. Assumes the boundary points to lie in front.
+    nb_boundary : unsigned int
+        The number of boundary points in the point cloud.
+    radius : positive float
+        Radius of the spherical outer boundary
+    healpy_order : unsigned int
+        The number of points on the outer boundary is defined as: 12*(healpy_order**2), as the healpy discretization of the sphere is used.
+    origin : 1 by 3 numpy vector of float
+        Contains x,y,z coordinates of center point of the sphere (by default [0.0,0.0,0.0]^T)
+
+    Returns
+    -------
+    remeshed_positions : numpy array of float
+        The positions of the points in the remeshed point cloud. The boundary points lie in front.
+    number of boundary points : unsigned int
+        The number of boundary points in the remeshed point cloud.
+    '''
+
+    radii2 = lambda r : np.sum(np.power(r-origin[np.newaxis, :], 2),axis=1)#function for computing the square of the radius
+    #clear points within inner boundary
+    #safety factor due to boundary hull being not a perfect sphere
+    max_angle = 45*2*np.pi/360.0 /healpy_order#max angle between two healpy directions
```

Comment:
I assume this is a safe overestimate?

Reply:
Yes, as the maximal angle for the 'healpy_order=1' discretization is 45°, so if I assume the angle to drop linearly with the healpy discretization order (using symmetry), this should be correct.